File: tests/CoreTests/GenericMethods/GenericMethods.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SharpKit.JavaScript;

namespace CoreTests.GenericMethods
{
    [JsType(JsMode.Prototype, IgnoreGenericMethodArguments=false)]
    class GenericMethods
    {
        void Foo()
        {
            var x = Create<GenericMethods>();
            var div = Create<DivElement>();
        }
        T Create<T>() where T : new()
        {
            return new T();
        }

    }
    [JsType(JsMode.Prototype, GenericArgumentJsCode="\"div\"")]
    class DivElement
    {
    }
}

File: tests/CoreTests/GenericMethods/GenericMethods.js

if (typeof(CoreTests) == "undefined")
    var CoreTests = {};
if (typeof(CoreTests.GenericMethods) == "undefined")
    CoreTests.GenericMethods = {};
CoreTests.GenericMethods.GenericMethods = function (){
};
CoreTests.GenericMethods.GenericMethods.prototype.Foo = function (){
    var x = this.Create(CoreTests.GenericMethods.GenericMethods);
    var div = this.Create("div");
};
CoreTests.GenericMethods.GenericMethods.prototype.Create = function (T){
    return new T();
};
CoreTests.GenericMethods.DivElement = function (){
};

